cargo-shuttle/src/lib.rs

```diff
@@ -658,6 +658,18 @@ impl Shuttle {
             err
         })?;
 
+        let child_stdout = runtime
+            .stdout
+            .take()
+            .context("child process did not have a handle to stdout")?;
+        let mut reader = BufReader::new(child_stdout).lines();
+        tokio::spawn(async move {
+            while let Some(line) = reader.next_line().await.unwrap() {
+                let mut stdout = stdout().lock();
+                writeln!(stdout, "{}", line).unwrap();
```

Comment:
Why not use `println!`?

Reply:
From `println` docs:

> The println! macro will lock the standard output on each call. If you call println! within a hot loop, this behavior may be the bottleneck of the loop. To avoid this, lock stdout with [io::stdout().lock()](https://doc.rust-lang.org/std/io/struct.Stdout.html)

https://doc.rust-lang.org/std/macro.println.html

Also, I believe using `stdout` makes the implementation more explicit.

Reply:
Ah, interesting. However, wouldn't the locking happening at each call of `println!` be the same as locking on `stdout inside the loop every time before the `writeln!`? I think the idea from the doc is to `lock` outside the loop and use the guard to write continuously to the stdout - meaning lock once for the entirety of the loop instead of locking on each iteration.

Reply:
>  I think the idea from the doc is to lock outside the loop and use the guard to write continuously to the stdout - meaning lock once for the entirety of the loop instead of locking on each iteration.

However, thinking better about this, we might need to write to the stdout of the cargo-shuttle process from different tasks too. I think this would block the other tasks.

Reply:
The speed of this loop is not super important either.

Reply:
@orhun what are your thoughts on this? Do you still find the `stdout` locking approach useful? We might be able to close this, I think this is the last unresolved comment.

Reply:
I'm okay with either way. If you're fine with `stdout` approach, feel free to give it a final review and we can proceed!

Reply:
Not sure what you mean by the final review (I did review the `stdout` approach), but I would remove the `stdout` and replace it with `println!` just because using it like you did they should be the same from my understanding. Correct me if I am wrong.

Reply:
14d7dfe27ae54d0d92a01d971bad8e884ff06dc5